sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesState.java

```diff
@@ -0,0 +1,91 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.util.SparseIntArray;
+import androidx.core.app.FrameMetricsAggregator;
+import io.sentry.protocol.MeasurementValue;
+import io.sentry.protocol.SentryId;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.WeakHashMap;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class ActivityFramesState {
+
+  private static final @NotNull ActivityFramesState instance = new ActivityFramesState();
+
+  private final @NotNull FrameMetricsAggregator frameMetricsAggregator =
+      new FrameMetricsAggregator();
+
+  private ActivityFramesState() {}
+
+  static @NotNull ActivityFramesState getInstance() {
+    return instance;
+  }
+
+  private final @NotNull Map<SentryId, Map<String, @NotNull MeasurementValue>>
+      activityMeasurements = Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
`m`: Why do we need a `WeakHashMap` here? Shouldn't a normal HashMap also be fine?